frontend/src/services/ApiService.tsx

```diff
@@ -0,0 +1,44 @@
+import ServiceProvider from "./ServiceProvider";
+import ProvidedServices from "./ProvidedServices";
+import axios from "axios";
+import { useEffect, useRef, useState } from "react";
+
+export interface IApiService {
+  login(user: string, pass: string): void;
+}
+
+const ApiServiceContext = ServiceProvider.createContext(
+  ProvidedServices.ApiService
+);
+export const useApiService = () =>
+  ServiceProvider.use<IApiService>(ProvidedServices.ApiService);
+
+export function ApiService({ children }: any) {
+  const [token, setToken] = useState("");
+
+  const loginUser = (user: string, pass: string) => {
+    axios
+      .post("http://localhost:5000/auth/login", {
+        username: user,
+        password: pass,
+      })
+      .then((response) => {
+        setToken(response.data.token);
```

Comment:
Check my most recent commit to roman/authentication-services - I added a Token service which should abstract some of the interaction with the local storage in the browser.